### PR TITLE
update to 6.1.0 - fix css path

### PR DESCRIPTION
**Pull request by frederikhors**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -6,14 +6,24 @@ const { encode, decode } = require('sourcemap-codec');
 const svelteHmr = require('./hmr');
 
 const {
-	major_version,
-	compile,
-	preprocess,
+	// major_version,
```
Comment (frederikhors) [inline index.js:9]: I think we do not need these anymore, am I wrong?

File: index.js [hunk 2]
```diff
@@ -62,15 +72,15 @@ function tryResolve(pkg, importer) {
...
-function exists(file) {
-	try {
-		fs.statSync(file);
-		return true;
-	} catch (err) {
-		if (err.code === 'ENOENT') return false;
-		throw err;
-	}
-}
+// function exists(file) {
```
Comment (frederikhors) [inline index.js:75]: Also this is useless if we use `fs.existsSync`, right?

**Comment by rixo** [comment]
We're already 2 breaking versions late on `rollup-plugin-svelte`. Things are starting to move fast... I've made a PR in the other direction, in the hope that `svelte-hmr` be integrated in the official plugin, and this one can be deprecated.

**Comment by zsparal** [comment]
Hi @rixo. As I was looking around in the devtool landscape when setting up a new Svelte project up came upon this project, alongside many others. As of now, I think that your "svelte-hmr" branch provides by far the best developer experience out of all the possible options:

1. Snowpack does not have proper source map support, especially for TypeScript. Some sourcemaps are generated but the lines do not match up correctly
2. Vite 2 with the NPM version of rollup-plugin-svelte-hot has the same problem
3. The official rollup-plugin-svelte has no HMR support
4. Vite 2 with rixo/rollup-plugin-svelte#svelte-hmr actually contains all the necessary features for this setup to work nicely. The only problem is that it's not an official release

It seems to me that the PR in the official svelte-plugin-rollup repo is not really moving forward. Would you maybe consider cutting a new (maybe even breaking) release for rollup-plugin-svelte-hot using the `svelte-hmr` branch? If there are any other issues with that proposal I'd be happy to help, even sending PRs and stuff. I think it's a pretty important building block for TypeScript users.

**Comment by rixo** [comment]
@Gustorn Hi! Sorry, I was sure I had answered that... Must have forgotten to push the submit button :sweat_smile: 

So... Yes, I can consider releasing the `svelte-hmr` branch.

Following discussion about the PR, it now seems that the best course of action would be to had a hook in the official plugin, and use it to just "inject" HMR from the hot fork. This way, we'd be totally forward compatible with future API changes, and the official plugin could remain agnostic in regard to HMR.

But if that's for Vite 2 that you want HMR support, there is a dedicated plugin that is in the work: https://www.npmjs.com/package/@svitejs/vite-plugin-svelte. It notably has advanced support for CSS-only updates, using specific Vite API.

On my side, I have also experimented with the minimal strict requirements for Vite support on the [vite branch](https://github.com/rixo/rollup-plugin-svelte-hot/tree/vite) of this repo. It shows that the current branch (the one submitted as a PR) is far from complete in regards to what's needed for Vite. I don't know what might still be missing on the new Vite branch...

Still, I'm not really sure of your intended usage... Do you still think it's worth doing a release before working on the "hook in the official plugin" plan?

**Comment by zsparal** [comment]
Thank you for the answer. Unfortunately the @svitejs plugin doesn't work for me yet, it fails when I'm using PostCSS + Tailwind in Svelte styles but I'll report it as a bug over there.

Hooking into the official plugin seems like a really good idea that would work quite well. The release itself is not that urgent for me, I just found that for now, the svelte-hmr branch worked very well with Vite 2, better than pretty much any of the alternatives out there so it would've been nice to have an official release (that's not tied to a GitHub branch existing).

**Comment by rixo** [comment]
@Gustorn Since the situation in regards to the official plugin might take a bit of time to decant, and considering in the interim the "hot" Rollup template and this plugin are becoming big entry points for confusion, this finished to convince me to do as you suggested.

I'm publishing the `svelte-hmr` branch as v1.0 of this plugin.

Currently, I am still assessing the quality of the branch before pushing the actual v1.0 to the registry. During this process, I'll do pre-releases on the `next` tag (i.e. `npm install rollup-plugin-svelte-hot@next`). I have already published 2 such versions, last one is `v1.0.0-1` currently (again, install that with the `@next` tag, because it might move fast).

It happens I also have a need to experiment with Vite support and, ideally, some level of predictability in the different Svelte plugins out there (for [Svench](https://github.com/rixo/svench)). And so, I have been working on fixing Vite support in the initial PR I submitted for HMR in the official plugin. This is all intended to ship with the aforementioned v1 of this plugin.

I'm very interested in your feedback, if you happen to experiment with the `next` version of this plugin in Vite.

@frederikhors If you're still using this plugin in Rollup based projects at the moment, I'd also be very interested in _your_ feedback in this regard -- don't we have big regressions in the Rollup/Nollup department with the recent change?

On another level, all this means we have a bit of docs updating to do before us, here and the hot template, as well as upgrade the hot template to v1 of this plugin. Any helping hand would be appreciated in this regard!

P.S. Just to be clear, this PR was closed automatically because I renamed master to a legacy v0 branch. I didn't realize this had had this effect before my attention was drawn back here by @Gustorn comment. Now that's still the proper outcome for this PR; we're never going to support v6... But we're well on our way to supporting v7/latest with the rebased master branch.

**Comment by frederikhors** [comment]
Dear @rixo,

in the last few days I tried Snowpack 3 and Vite.

Very difficult to be able to use them for more than one "hello world" project (although I have experience now!).

Especially for Snowpack it is still very immature and full of bugs (try using TailwindCss with Postcss if you can! Try to bundle with Rollup in esm!).

So I'm going back to your project which has been the most productive and effective so far.

In the next few hours I will tell you if there are any problems.

In the last project I remember that I didn't update the dependencies anymore as soon as I noticed this: https://github.com/rixo/svelte-template-hot/issues/21.

I would like to help you so that your work becomes a solid alternative for a concrete help for rapid application development!

**Comment by frederikhors** [comment]
I''m having troubles using this p.json:

```json
    "@rollup/plugin-commonjs": "17.1.0",
    "@rollup/plugin-node-resolve": "11.2.0",
    "@rollup/plugin-typescript": "8.2.0",
    "@tsconfig/svelte": "1.0.10",
    "nollup": "0.15.5",
    "rollup": "2.41.3",
    "rollup-plugin-css-only": "3.1.0",
    "rollup-plugin-hot": "0.1.1",
    "rollup-plugin-livereload": "2.0.0",
    "rollup-plugin-svelte-hot": "1.0.0-6",
    "rollup-plugin-terser": "7.0.2",
    "sirv-cli": "1.0.11",
    "svelte": "3.35.0",
    "svelte-check": "1.2.5",
    "svelte-preprocess": "4.6.9",
    "tslib": "2.1.0",
    "typescript": "4.2.3"
```

**Pull request closed without merging**